lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2470,6 +2470,15 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
         }
       }
     }
+
+    // @_staticExclusiveOnly types cannot be put into 'var's, only 'let'.
+    if (auto SD = VD->getInterfaceType()->getStructOrBoundGenericStruct()) {
+      if (SD->getAttrs().hasAttribute<StaticExclusiveOnlyAttr>() &&
```

Comment:
I think if you do this without checking the flag first even though the feature might not be enabled you'd get a diagnostic here.